packages/ozone/src/auth-verifier.ts

```diff
@@ -0,0 +1,218 @@
+import express from 'express'
+import * as ui8 from 'uint8arrays'
+import { IdResolver } from '@atproto/identity'
+import { AuthRequiredError, verifyJwt } from '@atproto/xrpc-server'
+
+type ReqCtx = {
+  req: express.Request
+}
+
+type RoleOutput = {
+  credentials: {
+    type: 'role'
+    isAdmin: boolean
+    isModerator: boolean
+    isTriage: true
+  }
+}
+
+type ModeratorOutput = {
+  credentials: {
+    type: 'moderator'
+    aud: string
+    iss: string
+    isAdmin: boolean
+    isModerator: boolean
+    isTriage: true
+  }
+}
+
+type StandardOutput = {
+  credentials: {
+    type: 'standard'
+    aud: string
+    iss: string
+    isAdmin: boolean
+    isModerator: boolean
+    isTriage: boolean
+  }
+}
+
+type NullOutput = {
+  credentials: {
+    type: 'null'
+    iss: null
+  }
+}
+
+export type AuthVerifierOpts = {
+  serviceDid: string
+  admins: string[]
+  moderators: string[]
+  triage: string[]
```

Comment:
I have a hunch we'll end-up wanting to toss these in the database before long, but seems like this is the right place to start 👍

Reply:
Yup yup agreed